x-pack/plugins/observability_solution/observability/public/components/custom_threshold/custom_threshold_rule_expression.test.tsx

```diff
@@ -164,7 +164,8 @@ describe('Expression', () => {
   it('should prefill the rule using the context metadata', async () => {
     const index = 'changedMockedIndex';
     const currentOptions: CustomThresholdPrefillOptions = {
-      alertOnGroupDisappear: false,
+      alertOnGroupDisappear: true,
```

Comment:
Could we get rid of the `alertOnGroupDisappear` and just use `alertOnNoData` for both with and without group? Since it's now mapped to a single checkbox it seems odd to keep two separate values on our schema. Or do we need to support this for backwards compatibility?

Reply:
Yes, we need to support this for backward compatibility. I've added this item to be considered for the v9 release.